src/OpenTelemetry.AutoInstrumentation.Native/pal.h

```diff
@@ -132,6 +136,52 @@ inline WSTRING GetCurrentModuleFileName()
     return EmptyWStr;
 }
 
+inline WSTRING GetModuleFilePath(const WSTRING& moduleName)
```

Comment:
As I understand, there are at least 3 possible layouts:
```
ZIP:
OTEL_HOME/<RID>/<native_profler_lib>
OTEL_HOME/net/OpenTelemetry.AutoInstrumentation.StartupHook.dll
```

```
NuGet,  platform dependent deployment:
<APP_FOLDER>/<native_profler_lib>
<APP_FOLDER>/OpenTelemetry.AutoInstrumentation.StartupHook.dll
```

```
NuGet,  platform independent deployment: (I suspect this is not validated correctly in IsStartupHookValid)
<APP_FOLDER>/runtimes/<RID>/<native_profler_lib>
<APP_FOLDER>/OpenTelemetry.AutoInstrumentation.StartupHook.dll
```

I'm also not sure, what is behaviour if OTEL_HOME envirinment variable not set, especially in case of zip deployment.

In any case, once we calculated the path - we should probe that file exists.

Reply:
Probably, we should apply manual add of profiler hook only if OpenTelemetry.AutoInstrumentation.StartupHook.dll is not present in environment variable at all. If it is there, but points to unexpected location, we should just log it.
I really hope that noone will rename the startup hook file :)